### ISSUE TITLE
Add a `beforeConstructCallback` in the constructor

### ISSUE THREAD

**Issue by gingerchew** [frankie-tech/element-plus#1]
This would let me hook in a call back before the construction step.

Needs to:

- pass in the selector
- save returned values to a variable
- pass said variable into `onConstructCallback` as an optional param

**Comment by gingerchew** [comment]
Adds initial version in commit **https://github.com/frankie-tech/element-plus/commit/908882183b3307fbf96f5c8cf1bd1bce9f25b62c**

Relevant notes:

```
normally the idea is for that to be hidden, but this can be necessary

need to make notes to urge against it, but say that it is allowed
```

I'd like for ElementPlus to do the necessary work first, and then run any extended code after that. The `beforeConstructCallback` should be seen as a necessary evil, not a first resort.

**Comment by gingerchew** [comment]
- [ ] change name to `onBeforeConstructCallback`

if the other one is `onConstructCallback` it's only fair

**Comment by gingerchew** [comment]
resolved in b0a109632666c3a6b82f72e86beee0438684209a
